### PR TITLE
Fix Decimal Sum Aggregate

### PR DESCRIPTION
**Pull request by majetideepak**
Use Varbinary as the intermediate type instead of LongDecimal.

Resolves: https://github.com/facebookincubator/velox/issues/3751
Resolves: https://github.com/facebookincubator/velox/issues/3931

### CODE REVIEW

File: velox/functions/prestosql/aggregates/SumAggregate.h [hunk 1]
```diff
@@ -227,7 +202,7 @@ bool registerSumAggregate(const std::string& name) {
           .integerVariable("a_precision")
           .integerVariable("a_scale")
           .argumentType("DECIMAL(a_precision, a_scale)")
-          .intermediateType("DECIMAL(38, a_scale)")
+          .intermediateType("VARBINARY")
```
Comment (Yuhta) [inline velox/functions/prestosql/aggregates/SumAggregate.h:205]: Is this compatible with Presto?

Comment (majetideepak) [reply velox/functions/prestosql/aggregates/SumAggregate.h:205]: @Yuhta it should be. See https://github.com/facebookincubator/velox/issues/3751
I am building a Prestissimo cluster with this change to verify as well. I will keep you posted here.

Comment (majetideepak) [reply velox/functions/prestosql/aggregates/SumAggregate.h:205]: @Yuhta The sum aggregate works via Prestissimo now.
```
presto> select sum(ss_sales_price) from hive.test.dec;
 _col0  
--------
 0.6848 
(1 row)
```

Comment (ebeworld) [reply velox/functions/prestosql/aggregates/SumAggregate.h:205]: Looks like presto intermediates are varbinary. https://code.byted.org/dp/prestissimo/blob/bd_main/presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/LongDecimalWithOverflowStateSerializer.java#L34

**Review by ebeworld — APPROVED** [review]

**Review by karteekmurthys — APPROVED** [review]

**Comment by majetideepak** [comment]
@Yuhta can you help review/merge this? Thanks.

**Pull request closed without merging**